DataDog/ansible-datadog
# Ansible-lint now runs on a collection instead of a role

The role create a collection at every pipeline to have a matching ansible-lint output with the collection. All variable has been renamed into the ``rolename_varname``  format.

## QA
  - CI :
    - CI now check ansible-lint as in a collection and as a role (in a role we have to exclude a few folder [see](https://github.com/DataDog/ansible-datadog/pull/490/files#diff-c2e20c78d6a1e9721a01e31217f318a569645d1a6f64d80f6425b4d63e36ecffR6) )
    - CI check the installation of the agent via ansible on multiple system 
    - CI **MUST** be green

  - Manual Testing : 
    - From the root of the git repository run ``./ci_test/collection-structure-setup.sh``. You should have no lint issues on the two ansible-lint check (one for the role, one for the collection). It will create a ``ansible_collections/`` folder containing the collection.
    - Go to the newly created ``ansible_collections/datadog/dd`` repository and run ``ansible-galaxy collection build``  it should generate a ``datadog-dd-1.0.0.tar.gz `` file.
    - Run locally at least one manual scenario locally (prefer the ``_full.yml``tests)

## QA Instructions

Follow and check that the previous process is working properly. Be careful that every option has been processed.
Additionally, install the agent on one or more ec2 instance to check the full ansible process. Preferably with a maximum of option in the playbook to ensure a maximum amount of variable are correctly acknowledged.
